docs/DEPLOYMENT.md

```diff
@@ -0,0 +1,195 @@
+efforts.
+
+### Daemonising the testground: a pre-requisite
+
+Up until November 2019, testground was operated by ephemeral CLI commands. There
+was no long-lived process. Testground was instantiated embedded in every single
+command. We were fleshing out the guts of the system, implementing the basic
+building blocks and primitives on top of which the rest of the system would
+hinge. Our focus was on the plumbing, not on the porcelain. This was by design;
+part of a structured SWE process to create a greenfield system.
+
+Testground has now matured significantly, and in order to implement the next set
+of features, we need the testground to operate as it was always intended to
+operate: as a long-lived daemon. Examples of such features include test run
+queueing & scheduling, test run state tracking, and GitHub automation.
+
+Daemonising the testground was already achieved in PR
+[#183](https://github.com/ipfs/testground/pull/183). All the deployment modes
+described herein rely on testground running as a daemon service.
+
+### Deploying testground
+
+Testground is a project that anyone can take and deploy in their own
+infrastructure, whether cloud, on-premise or even in a single-machine. It is not
+dependent on any PL-specific pieces of infrastructure. It is
+platform/provider-agnostic by design.
+
+That said, at Protocol Labs we're pursue continuous testing of the IPFS and
+libp2p projects (first). We thus operate Testground deployment as a service,
+watching GitHub, receiving commands, scheduling builds and runs on a fleet of
+workers, and tracking and reporting their progress and results via a visual
+dashboard (caveat: many of these pieces are WIP as r0 of this doc).
+
+The above entails managing infrastructure and environments. To automate such
+tasks, we've chosen a synergistic combination of popular tools:
+
+* *Terraform configurations* to manage the physical infrastructure. These are
+  _cloud-specific_. The PL team runs testground on AWS, and we've open sourced
+  our configurations.
+    * Users on AWS can borrow these configs and take off. Users using other
+      providers will need to replace the Terraform configs.
+* *Ansible playbooks* to manage the logical infrastructure, such as Docker
+  services, system services, overlay networks, etc. The Ansible scripts are
+  _cloud-leaky_. In theory they could be cloud-agnostic, but in practice various
+  items such as S3 storage, AWS ECR authentication, etc. are cloud-specific.
+    * Users wishing to deploy testground on an alternate provider, will need to
+      patch those elements.
+* _[terraform-inventory](https://github.com/adammck/terraform-inventory)_ for
+  the hand-off between the two.
+
+These assets handle the intantiation of various node types of nodes, and the
+backing infrastructural elements (e.g. machines, VPCs, security groups, etc.):
+
+1. **Testground master**: runs the daemon, watches GitHub, exposes an
```

Comment:
After the big brouhaha in the redis community about the master/slave naming, I think it's best to steer clear of "master" ... I was using "manager" (what Docker Swarm uses) but there are other words.